sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -78,11 +82,14 @@ trait NonLeafStatementExec extends CompoundStatementExec {
         case Array(field) if field.dataType == BooleanType =>
           df.limit(2).collect() match {
             case Array(row) => row.getBoolean(0)
-            case _ => false
+            case _ =>
+              throw QueryExecutionErrors.invalidBooleanStatementError(statement.getText)
           }
-        case _ => false
+        case _ =>
+          throw QueryExecutionErrors.invalidBooleanStatementError(statement.getText)
```

Comment:
I would maybe create two different error messages for these cases...
First one would be like "Boolean expression <expr> is invalid. It returns more than one row." and the second one would be similar to what you did here "Boolean statement is expected in the condition, but <expr> was found.".
This would better indicate what is the problem.